### PR TITLE
chore(ME-3430): policy v2 support

### PR DESCRIPTION
**Pull request by btoonk**
## Summary by CodeRabbit

- **New Features**
  - Introduced support for version 2 policies with enhanced permissions and conditions.
  - Added detailed permission structures for various services like databases, SSH, HTTP, TLS, VNC, RDP, and VPN.

- **Bug Fixes**
  - Updated policy test cases to include new versioned policy data, ensuring compatibility and correctness.

- **Documentation**
  - Updated policy documentation to reflect new fields and structures for v2 policies.

### CODE REVIEW

**Review by th3wingman — APPROVED** [review]
🚢

**Review by th3wingman — APPROVED** [review]
🚢

File: client/policy.go [hunk 1]
```diff
@@ -229,3 +254,70 @@ type PolicySocketAttachment struct {
 	Action string `json:"action" binding:"required"`
 	ID     string `json:"id" binding:"required"`
 }
+
+type PolicyPermissions struct {
```
Comment (adrianosela) [inline client/policy.go:258]: Missing godoc

**Review by adrianosela — COMMENTED** [review]
Missing godoc in a few places. Lgtm

File: client/policy.go [hunk 2]
```diff
@@ -229,3 +254,70 @@ type PolicySocketAttachment struct {
...
+	Database *DatabasePermissions `json:"database,omitempty"`
+	SSH      *SSHPermissions      `json:"ssh,omitempty"`
+	HTTP     *HTTPPermissions     `json:"http,omitempty"`
+	TLS      *TLSPermissions      `json:"tls,omitempty"`
+	VNC      *VNCPermissions      `json:"vnc,omitempty"`
+	RDP      *RDPPermissions      `json:"rdp,omitempty"`
+	VPN      *VPNPermissions      `json:"vpn,omitempty"`
+}
+
+type DatabasePermissions struct {
```
Comment (adrianosela) [inline client/policy.go:268]: Godoc

**Review by adrianosela — COMMENTED** [review]
Lgtm

**Pull request merged**